Mage.Sets/src/mage/cards/c/CaseOfTheBurningMasks.java

```diff
@@ -0,0 +1,167 @@
+package mage.cards.c;
+
+import java.util.HashSet;
+import java.util.Set;
+import java.util.UUID;
+
+import mage.MageObjectReference;
+import mage.abilities.Ability;
+import mage.abilities.common.CaseAbility;
+import mage.abilities.common.EntersBattlefieldTriggeredAbility;
+import mage.abilities.common.SimpleActivatedAbility;
+import mage.abilities.condition.Condition;
+import mage.abilities.costs.common.SacrificeSourceCost;
+import mage.abilities.effects.OneShotEffect;
+import mage.abilities.effects.common.DamageTargetEffect;
+import mage.cards.Card;
+import mage.cards.Cards;
+import mage.cards.CardsImpl;
+import mage.constants.Duration;
+import mage.constants.Outcome;
+import mage.constants.SubType;
+import mage.cards.CardImpl;
+import mage.cards.CardSetInfo;
+import mage.constants.CardType;
+import mage.constants.WatcherScope;
+import mage.constants.Zone;
+import mage.filter.StaticFilters;
+import mage.game.Game;
+import mage.game.events.GameEvent;
+import mage.players.Player;
+import mage.target.TargetCard;
+import mage.target.common.TargetCardInExile;
+import mage.target.common.TargetOpponentsCreaturePermanent;
+import mage.util.CardUtil;
+import mage.watchers.Watcher;
+
+/**
+ * Case of the Burning Masks {1}{R}{R}
+ * Enchantment - Case
+ * When this Case enters the battlefield, it deals 3 damage to target creature an opponent controls.
+ * To solve -- Three or more sources you controlled dealt damage this turn.
+ *
+ * @author DominionSpy
+ */
+public final class CaseOfTheBurningMasks extends CardImpl {
+
+    public CaseOfTheBurningMasks(UUID ownerId, CardSetInfo setInfo) {
+        super(ownerId, setInfo, new CardType[]{CardType.ENCHANTMENT}, "{1}{R}{R}");
+
+        this.subtype.add(SubType.CASE);
+
+        // When this Case enters the battlefield, it deals 3 damage to target creature an opponent controls.
+        Ability initialAbility = new EntersBattlefieldTriggeredAbility(
+                new DamageTargetEffect(3, "it"));
+        initialAbility.addTarget(new TargetOpponentsCreaturePermanent());
+        // To solve -- Three or more sources you controlled dealt damage this turn.
+        Condition toSolveCondition = new CaseOfTheBurningMasksCondition();
+        // Solved -- Sacrifice this Case: Exile the top three cards of your library. Choose one of them. You may play that card this turn.
+        Ability solvedAbility = new SimpleActivatedAbility(new CaseOfTheBurningMasksEffect(),
+                new SacrificeSourceCost());
+
+        CaseAbility ability = new CaseAbility(initialAbility, toSolveCondition, solvedAbility);
+        this.addAbility(ability, new CaseOfTheBurningMasksWatcher());
```

Comment:
Must add conditional card hint with solved status and other additional info for best UX.

Reply:
I've removed it from `PermanentImpl` and it's now being handled by the new `CaseSolvedHint`.